NextcloudTalk/SettingsTableViewController.swift

```diff
@@ -0,0 +1,676 @@
+
+    // MARK: User Profile
+
+    func refreshUserProfile() {
+        NCSettingsController.sharedInstance().getUserProfile { _ in
+            self.tableView.reloadData()
+        }
+        self.getActiveUserStatus()
+    }
+
+    func getActiveUserStatus() {
+        let activeAccount = NCDatabaseManager.sharedInstance().activeAccount()
+        NCAPIController.sharedInstance().getUserStatus(for: activeAccount) { userStatus, error in
+            if error == nil && userStatus != nil {
+                self.activeUserStatus = NCUserStatus(dictionary: userStatus!)
+                self.tableView.reloadData()
+            }
+        }
+    }
+
+    // MARK: Notifications
+
+    @objc func appStateHasChanged(notification: NSNotification) {
+        let appState = notification.userInfo!["appState"]
+        self.adaptInterfaceForAppState(appState: appState as? AppState ?? kAppStateReady)
+    }
+
+    @objc func contactsHaveBeenUpdated(notification: NSNotification) {
+        DispatchQueue.main.async {
+            self.tableView.reloadData()
+        }
+    }
+
+    @objc func contactsAccessHasBeenUpdated(notification: NSNotification) {
+        DispatchQueue.main.async {
+            self.tableView.reloadData()
+        }
+    }
+
+    // MARK: User Interface
+
+    func adaptInterfaceForAppState(appState: AppState) {
+        switch appState {
+        case kAppStateReady:
+            refreshUserProfile()
+        default:
+            break
+        }
+    }
+
+    // MARK: Profile actions
+
+    func userProfilePressed() {
+        let activeAccount = NCDatabaseManager.sharedInstance().activeAccount()
+        let userProfileVC = UserProfileViewController(account: activeAccount)
+        self.navigationController!.pushViewController(userProfileVC, animated: true)
+    }
+
+    // MARK: User Status
+
+    func presetnUserStatusOptions() {
+        let viewController = UserStatusTableViewController(userStatus: activeUserStatus!)
+        self.navigationController!.pushViewController(viewController, animated: true)
+    }
+
+    // MARK: User phone number
+
+    func checkUserPhoneNumber() {
+        NCSettingsController.sharedInstance().getUserProfile { _ in
+            let activeAccount = NCDatabaseManager.sharedInstance().activeAccount()
+            if activeAccount.phone.isEmpty {
+                self.presentSetPhoneNumberDialog()
+            }
+        }
+    }
+
+    func presentSetPhoneNumberDialog() {
+        let alertTitle = NSLocalizedString("Phone number", comment: "")
+        let alertMessage = NSLocalizedString("You can set your phone number so other users will be able to find you", comment: "")
+        let setPhoneNumberDialog = UIAlertController(title: alertTitle, message: alertMessage, preferredStyle: .alert)
+
+        setPhoneNumberDialog.addTextField { [self] textField in
+            let location = NSLocale.current.regionCode
+            textField.text = "+\(self.phoneUtil?.getCountryCode(forRegion: location) ?? 0)"
```

Comment:
Only set a country code if we could find one.
Setting "+0" would be confusing for the user.